spack-environments/archer2/compute-node/spack.yaml

```diff
@@ -0,0 +1,371 @@
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-aocc
+      - aocc/3.2.0
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: aocc@8.0.0
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags:
+        cflags: null
+        cxxflags: null
+        fflags: null
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-aocc
+      - aocc/8.0.0
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: aocc@8.1.0
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags:
+        cflags: null
+        cxxflags: null
+        fflags: null
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-aocc
+      - aocc/8.1.0
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: aocc@8.3.3
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags:
+        cflags: null
+        cxxflags: null
+        fflags: null
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-aocc
+      - aocc/8.3.3
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: cce@11.0.4
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-cray
+      - cce/11.0.4
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: cce@12.0.3
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-cray
+      - cce/12.0.3
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: cce@13.0.2
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-cray
+      - cce/13.0.2
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: gcc@8.1.0
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-gnu
+      - gcc/8.1.0
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: gcc@9.3.0
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-gnu
+      - gcc/9.3.0
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: gcc@10.2.0
+      paths:
+        cc: cc
+        cxx: CC
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-gnu
+      - gcc/10.2.0
+      environment: {}
+      extra_rpaths: []
```

Comment:
For GCC >=10, you will probably need to add:

```
fflags: -fallow-argument-mismatch
```

To allow MPI code to compile